### PR TITLE
bt-207: add onboard logic step 2

### LINKED ISSUE #207: Add logic to Onboarding Step 2
**User story:**
As a talent, I want to be able to choose and add badges to my profile, so that potential employers can get familiar with my soft skills

**Acceptance criteria:**
- “Your average project score”, “Your average lectures score”, “Level of English” should be chosen by default and talent can’t change it
- Talent should be able to choose only “Communication score”, “Working with team score”, “Punctuality” badges
- Talent should be able to choose several badges
- Talent should be able to go to the next step without choosing badges
- Talent should be able to back to the previous step "Profile" by clicking on the “Back” button
- Talent should be able to go to the next step "Skills and projects" by clicking on the “Next” button
- Talent 's information should be saved in the fields when switching between steps

### PR DESCRIPTION
**Pull request by mskmee**
All logic work as requirements, with out post actions

### CODE REVIEW

File: mobile/src/navigations/root/root.tsx [hunk 1]
```diff
@@ -26,27 +34,42 @@ const Root: React.FC = () => {
...
-              />
-          ))}
-
-      </RootStack.Navigator>
-  );
+                name={TalentOnboardingScreenName.SKILLS_AND_PROJECTS}
+                component={SkillsAndProjects}
+            />
+        </RootStack.Navigator>
+    );
```
Comment (likeri29) [inline mobile/src/navigations/root/root.tsx]: Why did you change our root navigation?

File: mobile/src/bundles/common/types/navigation/root-navigation-parameter-list.type.ts [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
 type RootNavigationParameterList = {
     [RootScreenName.AUTH_ROOT_ROUTE]: undefined;
     [RootScreenName.MAIN_ROOT_ROUTE]: undefined;
     [RootScreenName.ONBOARDING_ROOT_ROUTE]: undefined;
+    [TalentOnboardingScreenName.PROFILE]: undefined;
+    [TalentOnboardingScreenName.BSA_BADGES]: undefined;
+    [TalentOnboardingScreenName.SKILLS_AND_PROJECTS]: undefined;
+    [TalentOnboardingScreenName.CV_AND_CONTACTS]: undefined;
+    [TalentOnboardingScreenName.PREVIEW]: undefined;
 };
```
Comment (anogire) [inline mobile/src/bundles/common/types/navigation/root-navigation-parameter-list.type.ts]: We have this already in talent-onboarding-navigation-parameter-list-type.ts and others

File: mobile/src/navigations/root/root.tsx (hunk 1 above)
Comment (mskmee) [reply mobile/src/navigations/root/root.tsx]: I had add RootStack.Screens for navigations, but git shows than I have been deleted and add the same code as before, if there are a better way to add screens tell me please, I'll do it)
Or It's should be talanted stack for navigation?

**Review by illia-dobro — COMMENTED** [review]
'add navigate'  - change name of pull request

File: mobile/src/navigations/root/root.tsx (hunk 1 above)
Comment (likeri29) [reply mobile/src/navigations/root/root.tsx]: We have already talent onboarding navigation and in root we conditionally render it. And in that file we have all screens names and screens params etc.

Comment (mskmee) [reply mobile/src/navigations/root/root.tsx]: fix

File: mobile/src/bundles/common/types/navigation/root-navigation-parameter-list.type.ts (hunk 2 above)
Comment (mskmee) [reply mobile/src/bundles/common/types/navigation/root-navigation-parameter-list.type.ts]: fix

**Comment by katerynakostikova** [comment]
Where is the logic for step 2, form, validation and so on?

File: mobile/src/bundles/talent/talent-api.ts [hunk 3]
```diff
@@ -21,6 +24,10 @@ class TalentApi extends HttpApiBase {
         return payload;
     }
 
+    public completeBadgesStep(payload: BadgeStepDto): BadgeStepDto {
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/talent-api.ts:27]: api will be the same for all steps

File: mobile/src/bundles/talent/store/actions.ts [hunk 4]
```diff
@@ -14,4 +17,13 @@ const completeProfileStep = createAsyncThunk<
     return talentApi.completeProfileStep(profileStepPayload);
 });
 
-export { completeProfileStep };
+const completeBadgesStep = createAsyncThunk<
```
Comment (katerynakostikova) [inline mobile/src/bundles/talent/store/actions.ts:20]: Let's try to reuse completeProfileStep, I believe we can have only 1 action for each step

Comment (mskmee) [reply mobile/src/bundles/talent/store/actions.ts:20]: Tatiana make this changes in https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/341

**Review by katerynakostikova — APPROVED** [review]

**Review by anogire — APPROVED** [review]
